lib/pbio/drv/usb/stm32_usbd/usbd_pybricks.c

```diff
@@ -0,0 +1,512 @@
+                    (void)USBD_CtlSendData(pdev, (uint8_t *)hPybricks->data, req->wLength);
+                } else {
+                    hPybricks->CmdOpCode = req->bRequest;
+                    hPybricks->CmdLength = (uint8_t)req->wLength;
+
+                    (void)USBD_CtlPrepareRx(pdev, (uint8_t *)hPybricks->data, req->wLength);
+                }
+            }
+            break;
+
+        case USB_REQ_TYPE_VENDOR:
+            switch (req->bRequest)
+            {
+                case USBD_MS_VENDOR_CODE:
+                    (void)USBD_CtlSendData(pdev,
+                        (uint8_t *)USBD_OSDescSet,
+                        MIN(sizeof(USBD_OSDescSet), req->wLength));
+                    break;
+
+                default:
+                    USBD_CtlError(pdev, req);
+                    ret = USBD_FAIL;
+                    break;
+            }
+            break;
+
+        case USB_REQ_TYPE_STANDARD:
+            switch (req->bRequest)
+            {
+                case USB_REQ_GET_STATUS:
+                    if (pdev->dev_state == USBD_STATE_CONFIGURED) {
+                        (void)USBD_CtlSendData(pdev, (uint8_t *)&status_info, 2U);
+                    } else {
+                        USBD_CtlError(pdev, req);
+                        ret = USBD_FAIL;
+                    }
+                    break;
+
+                case USB_REQ_GET_INTERFACE:
+                    if (pdev->dev_state == USBD_STATE_CONFIGURED) {
+                        (void)USBD_CtlSendData(pdev, &ifalt, 1U);
+                    } else {
+                        USBD_CtlError(pdev, req);
+                        ret = USBD_FAIL;
+                    }
+                    break;
+
+                case USB_REQ_SET_INTERFACE:
+                    if (pdev->dev_state != USBD_STATE_CONFIGURED) {
+                        USBD_CtlError(pdev, req);
+                        ret = USBD_FAIL;
+                    }
+                    break;
+
+                case USB_REQ_CLEAR_FEATURE:
+                    break;
+
+                default:
+                    USBD_CtlError(pdev, req);
+                    ret = USBD_FAIL;
+                    break;
+            }
+            break;
+
+        default:
+            USBD_CtlError(pdev, req);
+            ret = USBD_FAIL;
+            break;
+    }
+
+    return (uint8_t)ret;
+}
+
+/**
+  * @brief  USBD_Pybricks_GetCfgDesc
+  *         return configuration descriptor
+  * @param  length : pointer data length
+  * @retval pointer to descriptor buffer
+  */
+static uint8_t *USBD_Pybricks_GetCfgDesc(uint16_t *length) {
+    *length = (uint16_t)sizeof(USBD_Pybricks_CfgDesc);
+    return USBD_Pybricks_CfgDesc;
+}
+
+/**
+  * @brief  USBD_Pybricks_DataIn
+  *         Data sent on non-control IN endpoint
+  * @param  pdev: device instance
+  * @param  epnum: endpoint number
+  * @retval status
+  */
+static uint8_t USBD_Pybricks_DataIn(USBD_HandleTypeDef *pdev, uint8_t epnum) {
```

Comment:
```suggestion
static USBD_StatusTypeDef USBD_Pybricks_DataIn(USBD_HandleTypeDef *pdev, uint8_t epnum) {
```

Then we can avoid casting return values.

Reply:
Same comment applies to other functions.

Reply:
`USBD_ClassTypeDef` uses `uint8_t` for the return type, so I think it's preferable to match that, do you agree?

Reply:
No, the STM32 libraries do a lot of things I don't agree with. :grin:

I don't think we need to fix everything we don't agree with, but removing all of the unnecessary casts seems like an easy way to make the code easier to read.

I'm happy to do these kinds of cleanups later though if you don't want to spend time on it.

Reply:
Got it. I didn't think it would actually compile without a cast somewhere else, but maybe the compiler I'm used to using are less forgiving. This does work so I went ahead and made the change.